docs/generate-oval.py

```diff
@@ -0,0 +1,147 @@
+    }.items():
+        oval = ET.SubElement(generator, key)
+        oval.text = value
+
+    definitions = ET.SubElement(oval_definitions, "definitions")
+    for item in items:
+
+        definition = ET.SubElement(definitions, "definition")
+        definition.set("class", "patch")
+        definition.set("id", "oval:{}:def:1".format(item["id"]))
+        definition.set("version", "1")
+
+        metadata = ET.SubElement(definition, "metadata")
+
+        ET.SubElement(metadata, "title").text = item["name"]
+
+        affected = ET.SubElement(metadata, "affected")
+        affected.set("family", "unix")
+        ET.SubElement(affected, "platform").text = "All"  # is this valid?
+
+        if "issues" in item:
+            for issue in item["issues"]:
+                reference = ET.SubElement(metadata, "reference")
+                reference.set("ref_id", issue)
+                if issue.startswith("CVE-"):
+                    reference.set(
+                        "ref_url", "https://nvd.nist.gov/vuln/detail/{}".format(issue)
+                    )
+                    reference.set("source", "CVE")
+
+        if "description" in item:
+            ET.SubElement(metadata, "description").text = "\n".join(item["description"])
+
+        criteria = ET.SubElement(definition, "criteria")
+        criteria.set("operator", "OR")
+        criterion = ET.SubElement(criteria, "criterion")
+        criterion.set("comment", item["name"])
+        criterion.set("test_ref", "oval:{}:tst:1".format(item["id"]))
+
+    tests = ET.SubElement(oval_definitions, "tests")
+    for item in items:
+
+        red_def = ET.SubElement(tests, "red-def:fwupdsecattr_test")
+        red_def.set("check", "at least one")
+        red_def.set("comment", item["name"])
+        red_def.set("id", "oval:{}:tst:1".format(item["id"]))
+        red_def.set("version", "1")
+
+        red_def_object = ET.SubElement(red_def, "red-def:object")
+        red_def_object.set("red-def:object", "oval:{}:obj:1".format(item["id"]))
+        red_def_state = ET.SubElement(red_def, "red-def:state")
+        red_def_state.set("red-def:state", "oval:{}:ste:1".format(item["id"]))
+
+    objects = ET.SubElement(oval_definitions, "objects")
+    for item in items:
+
+        red_def = ET.SubElement(objects, "red-def:fwupdsecattr_object")
+        red_def.set("id", "oval:{}:obj:1".format(item["id"]))
+        red_def.set("version", "1")
+
+        red_def_stream_id = ET.SubElement(red_def, "red-def:fwupdsecattr_object")
+        red_def_stream_id.set("datatype", "string")
+        red_def_stream_id.text = format(item["id"])
+
+    states = ET.SubElement(oval_definitions, "states")
+    for item in items:
+
+        red_def = ET.SubElement(states, "red-def:fwupdsecattr_state")
+        red_def.set("id", "oval:{}:ste:1".format(item["id"]))
+        red_def.set("version", "1")
+
+        red_def_security_attr = ET.SubElement(red_def, "red-def:fwupdsecattr_object")
```

Comment:
same here. red-def should be "red-def:**output-key-name**" in the "state" element.

for example:
<red-def:**security-attr** datatype="string" operation="pattern match">not-found|not-enabled</red-def:security-attr>

Reply:
Fixed, thanks.